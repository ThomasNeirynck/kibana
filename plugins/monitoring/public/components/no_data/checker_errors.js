import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  EuiTitle,
  EuiTextColor,
  EuiIcon,
  EuiFlexGroup,
  EuiFlexItem
} from '@elastic/eui';

const ErrorList = ({ errors }) => {
  return errors.map((error, errorIndex) => {
    const { message, statusCode, error: friendlyName } = error;
    return (
      <div key={`checker-error-${errorIndex}`}>
        <EuiTextColor color="danger">
          <EuiFlexGroup alignItems="center" gutterSize="s">
            <EuiFlexItem grow={false}>
              <EuiIcon type="alert" color="danger" aria-label="alert" />
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              {statusCode} {friendlyName}
            </EuiFlexItem>
            <EuiFlexItem grow={false}>{message}</EuiFlexItem>
          </EuiFlexGroup>
        </EuiTextColor>
      </div>
    );
  });
};

export function CheckerErrors(props) {
  if (props.errors === undefined || props.errors.length === 0) {
    return null;
  }

  return (
    <Fragment>
      <EuiTitle size="s">
        <h2>Errors</h2>
      </EuiTitle>

      <p>
        There were some errors encountered in trying to check Elasticsearch
        settings. You need administrator rights to check the settings and, if
        needed, to enable the monitoring collection setting.
      </p>

      <ErrorList {...props} />
    </Fragment>
  );
}

CheckerErrors.propTypes = {
  errors: PropTypes.array
};
