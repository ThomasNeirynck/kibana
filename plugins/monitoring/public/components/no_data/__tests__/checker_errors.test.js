import React from 'react';
import { wrap, forbidden } from 'boom';
import { render } from 'enzyme';
import { CheckerErrors } from '../checker_errors';

describe('CheckerErrors', () => {
  test('should render nothing if errors is empty', () => {
    const component = render(<CheckerErrors errors={[]} />);
    expect(component).toMatchSnapshot();
  });

  test('should render typical boom errors from api response', () => {
    const err1 = forbidden(new Error('no access for you'));
    const err2 = wrap(new Error('bad thing happened'));
    const errors = [err1, err2].map(err => err.output.payload);
    const component = render(<CheckerErrors errors={errors} />);
    expect(component).toMatchSnapshot();
  });
});
