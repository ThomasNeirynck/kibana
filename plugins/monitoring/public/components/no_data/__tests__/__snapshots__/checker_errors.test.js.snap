// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckerErrors should render nothing if errors is empty 1`] = `null`;

exports[`CheckerErrors should render typical boom errors from api response 1`] = `
Array [
  <h2
    class="euiTitle euiTitle--small"
  >
    Errors
  </h2>,
  <p>
    There were some errors encountered in trying to check Elasticsearch settings. You need administrator rights to check the settings and, if needed, to enable the monitoring collection setting.
  </p>,
  <div>
    <span
      class="euiTextColor euiTextColor--danger"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-label="alert"
            class="euiIcon euiIcon--medium euiIcon--danger"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill-rule="evenodd"
            >
              <path
                d="M7.5 2.236L1.618 14h11.764L7.5 2.236zm.894-.447l5.882 11.764A1 1 0 0 1 13.382 15H1.618a1 1 0 0 1-.894-1.447L6.606 1.789a1 1 0 0 1 1.788 0z"
              />
              <path
                d="M7 6h1v5H7zM7 12h1v1H7z"
              />
            </g>
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          403 Forbidden
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          Error: no access for you
        </div>
      </div>
    </span>
  </div>,
  <div>
    <span
      class="euiTextColor euiTextColor--danger"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <svg
            aria-label="alert"
            class="euiIcon euiIcon--medium euiIcon--danger"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill-rule="evenodd"
            >
              <path
                d="M7.5 2.236L1.618 14h11.764L7.5 2.236zm.894-.447l5.882 11.764A1 1 0 0 1 13.382 15H1.618a1 1 0 0 1-.894-1.447L6.606 1.789a1 1 0 0 1 1.788 0z"
              />
              <path
                d="M7 6h1v5H7zM7 12h1v1H7z"
              />
            </g>
          </svg>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          500 Internal Server Error
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          An internal server error occurred
        </div>
      </div>
    </span>
  </div>,
]
`;
